scte35/state.go

```diff
@@ -211,8 +211,13 @@ func (s *state) Close(desc SegmentationDescriptor) ([]SegmentationDescriptor, er
 		d := s.open[i]
 		closed = append(closed, d)
 		if desc.Equal(d) {
-			// found our descriptor, remove it and everything after it
-			s.open = s.open[0:i]
+			// found our descriptor at index i, remove it
+			// Shift s.open left by one index.
+			copy(s.open[i:], s.open[i+1:])
+			// Delete last element
+			s.open[len(s.open)-1] = nil
+			// Truncate slice
+			s.open = s.open[:len(s.open)-1]
 			return closed, nil
```

Comment:
I think we will need to modify closed, since we are no longer closing all signals in the slice after the one we found.

Reply:
oh yeah good point, updating it.